### PR TITLE
Feature/plcn 360 create update user command handler

### PR DESCRIPTION
**Pull request by Nicolai-Bak**
Added handler to update email and name on user entity. Fails if user not found or if email is already used by another user.

### CODE REVIEW

File: src/Pelican.Application/Users/Commands/UpdateUser/UpdateUserCommandHandler.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	}
+
+	public async Task<Result<User>> Handle(
+		UpdateUserCommand command,
+		CancellationToken cancellationToken)
+	{
+		User? user = await _unitOfWork
+			.UserRepository
+			.FirstOrDefaultAsync(
+				expression: u => u.Id == command.User.Id,
```
Comment (AntonEsbjergIT-Minds) [inline src/Pelican.Application/Users/Commands/UpdateUser/UpdateUserCommandHandler.cs:24]: Named argument specifications seem unnecessary

File: src/Pelican.Application/Users/Commands/UpdateUser/UpdateUserCommandHandler.cs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+				cancellationToken: cancellationToken);
+
+		if (user is null)
+		{
+			return new Error(
+				code: "User.NotFound",
+				message: $"{command.User.Id} was not found");
```
Comment (AntonEsbjergIT-Minds) [inline src/Pelican.Application/Users/Commands/UpdateUser/UpdateUserCommandHandler.cs:31]: This message doesn't seems that useful. You are returning "{guid} was not found" Perhaps change it to sayre "User with id: {Guid} was not found" or "Id for user {user.Name} was not found"

File: src/Pelican.Application/Users/Commands/UpdateUser/UpdateUserCommandHandler.cs [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+				.AnyAsync(
+					expression: x => x.Email == command.User.Email,
+					cancellationToken: cancellationToken))
+		{
+			return new Error(
+				code: "Email.InUse",
+				message: "Email already in use");
+		}
+
+		user.Name = command.User.Name ?? user.Name;
```
Comment (AntonEsbjergIT-Minds) [inline src/Pelican.Application/Users/Commands/UpdateUser/UpdateUserCommandHandler.cs:45]: You lack a test for the branch where command.User.Name is null

File: src/Pelican.Application/Users/Commands/UpdateUser/UpdateUserCommandHandler.cs [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+		user.Email = command.User.Email ?? user.Email;
```
Comment (AntonEsbjergIT-Minds) [inline src/Pelican.Application/Users/Commands/UpdateUser/UpdateUserCommandHandler.cs:46]: You lack a test for the branch where command.User.Email is null

File: test/Pelican.Application.Test/Users/Commands/UpdateUser/UpdateUserCommandHandlerTests.cs [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+			expression: x => x
+				.UserRepository
+				.AnyAsync(
+					u => u.Email == command.User.Email,
+					default),
+			times: Times.Once);
+	}
+
+	[Fact]
+	public async void Handle_UserUpdated_ReturnFailure()
```
Comment (AntonEsbjergIT-Minds) [inline test/Pelican.Application.Test/Users/Commands/UpdateUser/UpdateUserCommandHandlerTests.cs:126]: Name should be Handle_UserUpdated_ReturnsSuccess()

File: src/Pelican.Application/Users/Commands/UpdateUser/UpdateUserCommandHandler.cs [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+				cancellationToken: cancellationToken);
+
+		if (user is null)
+		{
+			return new Error(
+				code: "User.NotFound",
+				message: $"{command.User.Id} was not found");
+		}
+
+		if (command.User.Email is not null && await _unitOfWork
```
Comment (AntonEsbjergIT-Minds) [inline src/Pelican.Application/Users/Commands/UpdateUser/UpdateUserCommandHandler.cs:34]: Add a test where Email is null

**Review by AntonEsbjergIT-Minds — CHANGES_REQUESTED** [review]

File: src/Pelican.Application/Users/Commands/UpdateUser/UpdateUserCommandHandler.cs (hunk 4 above)
Comment (Nicolai-Bak) [reply src/Pelican.Application/Users/Commands/UpdateUser/UpdateUserCommandHandler.cs:46]: branch removed

File: src/Pelican.Application/Users/Commands/UpdateUser/UpdateUserCommandHandler.cs (hunk 3 above)
Comment (Nicolai-Bak) [reply src/Pelican.Application/Users/Commands/UpdateUser/UpdateUserCommandHandler.cs:45]: branch removed

File: src/Pelican.Application/Users/Commands/UpdateUser/UpdateUserCommandHandler.cs (hunk 1 above)
Comment (Nicolai-Bak) [reply src/Pelican.Application/Users/Commands/UpdateUser/UpdateUserCommandHandler.cs:24]: removed

File: src/Pelican.Application/Users/Commands/UpdateUser/UpdateUserCommandHandler.cs (hunk 6 above)
Comment (Nicolai-Bak) [reply src/Pelican.Application/Users/Commands/UpdateUser/UpdateUserCommandHandler.cs:34]: branch removed as validator does not allow email to be null

**Review by AntonEsbjergIT-Minds — APPROVED** [review]

**Review by itm-mne — APPROVED** [review]
